arch/arm/core/aarch32/Kconfig

```diff
@@ -76,6 +76,38 @@ config ISA_ARM
 	  processor start-up. Much of its functionality was subsumed into T32 with
 	  the introduction of Thumb-2 technology.
 
+config ASSEMBLER_ISA_THUMB2
+	bool
```

Comment:
this is hidden option, so the value will be based on the settings below
now, for ARMV7_R both ISA_ARM and ISA_THUMB2 are selected (also hidden options), so isn't this always unset? This is somewhat explained in the last sentence; i wonder if it can be made more clear. 
I would remove the depends on and make the default sentence more explicit

Reply:
The intention is as follows:
* For Cortex-M, since only `ISA_THUMB2` is selected, `ASSEMBLER_ISA_THUMB2=y`.
* For Cortex-A/-R, both `ISA_THUMB2` and `ISA_ARM` are selected, `ASSEMBLER_ISA_THUMB2=n`.

If we remove `depends on !ISA_ARM`, we will have `ASSEMBLER_ISA_THUMB2=y` on the Cortex-A/-R, which is not what we want.

Reply:
I suggested to make it: `default y if ISA_THUMB2 && !ISA_ARM`

Reply:
While that would achieve the same apparent result, it does not prevent `ASSEMBLER_ISA_THUMB2` from being selected when `ISA_ARM=y`; of course, one could argue that `ASSEMBLER_ISA_THUMB2` is a helper symbol and was never meant to be selected, but it is still nice to have a safe-guard against an invalid configuration.

Reply:
Yes, exactly, I wanted to promote ease-of-reading against fully-secure definition that is hard to read. I do not expect any option to select ASSEMBLER_ISA_THUMB2. But I ve got no strong opinion about it, you can leave it as is if you think it is better.

Reply:
I think having both could be beneficial, in terms of semantics.

```
	default y if ISA_THUMB2 && !ISA_ARM
	depends on !ISA_ARM
```

The first line conveys the full meaning, while the second line prohibits (potential) invalid configurations.

I will push a new commit that addresses this.